File: app/src/main/java/com/proxy/shadowsocksr/impl/plugin/obfs/ObfsChooser.java
package com.proxy.shadowsocksr.impl.plugin.obfs;

public final class ObfsChooser
{
    public static AbsObfs getObfs(String obfsMethod, String rmtIP, int rmtPort,
            int tcpMss, String usrParamStr)
    {
        switch (obfsMethod)
        {
        case "http_simple":
            return new HttpSimpleObfs(usrParamStr, rmtIP, rmtPort, tcpMss);
        case "origin":
        default:
            return new SSObfs(usrParamStr, rmtIP, rmtPort, tcpMss);
        }
    }
}

File: app/src/main/java/com/proxy/shadowsocksr/impl/plugin/obfs/ObfsChooser.kt
package com.proxy.shadowsocksr.impl.plugin.obfs

class ObfsChooser
{
    companion object
    {
        fun getObfs(obfsMethod: String, rmtIP: String, rmtPort: Int,
                    tcpMss: Int, usrParamStr: String): AbsObfs
        {
            when (obfsMethod)
            {
                "http_simple" -> return HttpSimpleObfs(usrParamStr, rmtIP, rmtPort, tcpMss)
            //... -> ...
                else -> return SSObfs(usrParamStr, rmtIP, rmtPort, tcpMss)//default origin
            }
        }
    }
}
